<?xml version="1.0"?>
<robot name="gofa_with_carrier">

  <!-- Robot end-effector link -->
  <link name="panda_hand"/>

  <!-- Carrier -->
  <link name="carrier">
    <visual>
      <geometry>
        <mesh filename="package://gofa_viz/resource/carrier_meshes/carrier_full.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://gofa_viz/resource/carrier_meshes/carrier_full.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <!-- Fixed joint connecting carrier to robot hand -->
  <joint name="hand_to_carrier" type="fixed">
    <parent link="panda_hand"/>
    <child link="carrier"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>

</robot>
